clients/samples/gemm_alphavec/sample_hipblaslt_gemm_alphavec_ext.cpp

```diff
@@ -0,0 +1,147 @@
+                              runner.beta,
+                              runner.d_a,
+                              runner.d_b,
+                              runner.d_c,
+                              runner.d_d,
+                              runner.d_alphaVec,
+                              runner.d_workspace,
+                              runner.max_workspace_size,
+                              runner.stream);
+    });
+
+    return 0;
+}
+
+void simpleGemmAlphaVecExt(hipblasLtHandle_t  handle,
+                           hipblasOperation_t trans_a,
+                           hipblasOperation_t trans_b,
+                           int64_t            m,
+                           int64_t            n,
+                           int64_t            k,
+                           int64_t            batch_count,
+                           float&             alpha,
+                           float&             beta,
+                           void*              d_a,
+                           void*              d_b,
+                           void*              d_c,
+                           void*              d_d,
+                           void*              d_alphaVec,
+                           void*              d_workspace,
+                           int64_t            max_workspace_size,
+                           hipStream_t        stream)
+{
+    hipblaslt_ext::GemmPreference gemmPref;
+    gemmPref.setMaxWorkspaceBytes(max_workspace_size);
+    hipblaslt_ext::Gemm gemm(handle,
+                             trans_a,
+                             trans_b,
+                             HIPBLASLT_R_16F,
+                             HIPBLASLT_R_16F,
+                             HIPBLASLT_R_16F,
+                             HIPBLASLT_R_16F,
+                             HIPBLASLT_COMPUTE_F32);
+
+    hipblaslt_ext::GemmEpilogue
+        epilogue; // No action needed, default is HIPBLASLT_EPILOGUE_DEFAULT. (Gemm only)
+    hipblaslt_ext::GemmInputs inputs;
+    inputs.a             = d_a;
+    inputs.b             = d_b;
+    inputs.c             = d_c;
+    inputs.d             = d_d;
+    inputs.alpha         = &alpha;
+    inputs.beta          = &beta;
+    inputs.scaleAlphaVec = d_alphaVec; // use AlphaVec mode
+    gemm.setProblem(m, n, k, batch_count, epilogue, inputs);
+
+    const int                                     request_solutions = 1;
+    std::vector<hipblasLtMatmulHeuristicResult_t> heuristicResult;
+    CHECK_HIPBLASLT_ERROR(gemm.algoGetHeuristic(request_solutions, gemmPref, heuristicResult));
+
+    if(heuristicResult.empty())
+    {
+        std::cerr << "No valid solution found!" << std::endl;
+        return;
+    }
+
+    // In this sample, the workspace is already allocated with max_workspace_size
+    // If not, calculate the needed workspace_size and allocate d_workspace here
+    // uint64_t workspace_size = 0;
+    // for(int i = 0; i < returnedAlgoCount; i++)
```

Comment:
add check if the d_worksapce size is biger enough for returned algo

Reply:
Fixed